src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -271,3 +293,26 @@ func onBytesGiven(destAddr string, req *http.Request, bytes int64) {
 
 	}
 }
+
+func hostFn(fqdns map[string]string) func(*http.Request) string {
+	// We prefer to use the fronting provider through which we have been
+	// reached, because we expect that to be unblocked, but if something goes
+	// wrong (e.g. in old give mode peers) we'll use just any configured host.
+	return func(req *http.Request) string {
+		for provider, fn := range frontingProviders {
+			if fn(req) {
+				return fqdns[provider]
+			}
+		}
+		// We don't know about this provider... for backwards
+		// compatibility, let's try just any of the supplied FQDNs.
+		log.Debugf("Falling back to just any FQDN")
+		for _, fqdn := range fqdns {
+			return fqdn
+		}
+		// should never be reached, because the client is checking for empty
+		// `-frontfqdn`s, but Go doesn't know that and insists we must return
+		// something here.
+		return ""
```

Comment:
@oxtoacart : I don't think it is possible since we'd have exited the program before ever starting to listen for client connections.  I've just added a github comment where I'm checking for this.  This is also why I added the editorial comment; if I believed this could be reached, some noisier error handling might be warranted here.